### ISSUE TITLE
[RovingFocusGroup] Incorrectly applies `aria-orientation` to some roles

### ISSUE THREAD

**Issue by jjenzz** [radix-ui/primitives#964]
Originally [reported in discord](https://discord.com/channels/752614004387610674/803656530259738674/907358269339738173) when using `ToggleGroup`'s `orientation` prop.

The only roles that `aria-orientation` should be supplied to are as follows:
https://www.w3.org/TR/wai-aria/#aria-orientation
